langchain/llms/anyscale.py

```diff
@@ -0,0 +1,113 @@
+from langchain.utils import get_from_dict_or_env
+import requests
+
+class AnyscaleLLM(LLM):
+    """Wrapper around Anyscale Services.
+    To use, you should have the environment variable ``ANYSCALE_SERVICE_URL`` set with your Anyscale Service URL, or pass it as a named parameter to the constructor.
+    
+    Example:
+        .. code-block:: python
+            from langchain.llms import AnyscaleLLM
+            anyscalellm = AnyscaleLLM(anyscale_service_url="SERVICE_URL",
+                                anyscale_service_route="SERVICE_ROUTE",
+                                anyscale_service_token="SERVICE_TOKEM")
+            # To leverage Ray for parallel processing
+	    import ray
+	    prompt_list=[]
+            @ray.remote
+            def send_query(llm, prompt):
+                resp = llm(text)
+                return resp
+            futures = [send_query.remote(anyscalellm, prompt) for promt in prompt_list]
+            results = ray.get(futures)
+    """
+
+    client: Any  #: :meta private:
+    model_kwargs: Optional[dict] = None
+    """Key word arguments to pass to the model."""
+
+    anyscale_service_url: Optional[str] = None
+    anyscale_service_route: Optional[str] = None
+    anyscale_service_token: Optional[str] = None
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key and python package exists in environment."""
+        anyscale_service_url = get_from_dict_or_env(
+            values, "anyscale_service_url", "ANYSCALE_SERVICE_URL"
+        )
+        anyscale_service_route = get_from_dict_or_env(
+            values, "anyscale_service_route", "ANYSCALE_SERVICE_ROUTE"
+        )
+        anyscale_service_token = get_from_dict_or_env(
+            values, "anyscale_service_token", "ANYSCALE_SERVICE_TOKEN"
+        )
+        try:
+            anyscale_service_endpoint = f'{anyscale_service_url}/-/route'
+            headers = {'Authorization': f'Bearer {anyscale_service_token}'}
+            resp = requests.get(anyscale_service_endpoint, headers=headers)
+        except requests.exceptions.RequestException as e:  # This is the correct syntax
+            raise ValueError(e)
+        values['anyscale_service_url'] = anyscale_service_url
+        values['anyscale_service_route'] = anyscale_service_route
+        values['anyscale_service_token'] = anyscale_service_token
+        return values
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        """Get the identifying parameters."""
+        return {
+            "anyscale_service_url": self.anyscale_service_url,
+            "anyscale_service_route": self.anyscale_service_route
+            #**{"model_kwargs": _model_kwargs},
+        }
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return "Anyscale LLM"
```

Comment:
convention so far as has been to use kebab-case, and probably don't need LLM in there. so really just "anyscale"

Reply:
Updated. Also modified class name just to be "Anyscale" to follow the convention